gdx/src/com/badlogic/gdx/graphics/glutils/IndexArray.java

```diff
@@ -40,12 +49,12 @@ public IndexArray (int maxIndices) {
 
 	/** @return the number of indices currently stored in this buffer */
 	public int getNumIndices () {
-		return buffer.limit();
+		return empty ? 0 : buffer.limit();
 	}
 
 	/** @return the maximum number of indices this IndexArray can store. */
 	public int getNumMaxIndices () {
-		return buffer.capacity();
+		return empty ? 0 : buffer.capacity();

```

Comment:
Would the previous simple "return buffer.limit();" not suffice if we, as you suggest, set the buffer's limit to "0" after "fake creating" it with a size>0